tests/host_reference/clover_reference.cpp

```diff
@@ -312,18 +312,19 @@ void tmc_dslash(void *out, void **gauge, void *in, void *clover, void *cInv, dou
   void *tmp1 = safe_malloc(Vh * spinor_site_size * precision);
   void *tmp2 = safe_malloc(Vh * spinor_site_size * precision);
 
-  if (dagger) {
-    twistCloverGamma5(tmp1, in, clover, cInv, dagger, kappa, mu, flavor, 1 - parity, QUDA_TWIST_GAMMA5_INVERSE,
-                      precision);
-    if (matpc_type == QUDA_MATPC_EVEN_EVEN_ASYMMETRIC || matpc_type == QUDA_MATPC_ODD_ODD_ASYMMETRIC) {
-      wil_dslash(tmp2, gauge, tmp1, parity, dagger, precision, param);
-      twistCloverGamma5(out, tmp2, clover, cInv, dagger, kappa, mu, flavor, parity, QUDA_TWIST_GAMMA5_INVERSE, precision);
-    } else {
-      wil_dslash(out, gauge, tmp1, parity, dagger, precision, param);
-    }
-  } else {
+
+  if (matpc_type == QUDA_MATPC_EVEN_EVEN_ASYMMETRIC || matpc_type == QUDA_MATPC_ODD_ODD_ASYMMETRIC) {
```

Comment:
I noticed that these two tests were failing
```
mpirun -n 1 ./tests/dslash_test  \
        --dslash-type twisted-clover --compute-clover 1 \
		--matpc even-even-asym \
	   	--dim  4 4 4 4 --prec ${prec} \
		--gridsize 1 1 1 1\
		--niter 2 \
		--compute-clover-trlog 1 \
		--dagger \
 		--gtest_output=xml:gauge_path_test_${prec}.xml | tee test-even-even-asym-dag.txt
```
```
mpirun -n 1 ./tests/dslash_test  \
        --dslash-type twisted-clover --compute-clover 1 \
		--matpc odd-odd-asym \
	   	--dim  4 4 4 4 --prec ${prec} \
		--gridsize 1 1 1 1\
		--niter 2 \
		--compute-clover-trlog 1 \
		--dagger \
 		--gtest_output=xml:gauge_path_test_${prec}.xml | tee test-odd-odd-asym-dag.txt
```
with
```
1.000000e-01 Failures: 0 / 3072  = 0.000000e+00
1.000000e-02 Failures: 2010 / 3072  = 6.542969e-01
1.000000e-03 Failures: 2944 / 3072  = 9.583333e-01
1.000000e-04 Failures: 3060 / 3072  = 9.960938e-01
``` 
this change should fix those two tests. For completeness, I try all combinations of 
```
--matpc even-even odd-odd even-even-asym odd-odd-asym
```
with the `--dagger` option on and off

Reply:
The Reference implementation of  `computeTMCloverForceQuda` works when tested with
```
mpirun -n 2  ./tests/TMCloverForce_test  --dslash-type twisted-clover --compute-clover 1 \
        --matpc even-even-asym   --dim  4 4 4 4 --prec double   --gridsize 2 1 1 1\
        --niter 1  --compute-clover-trlog 1  --verbosity verbose  --kappa 1  --mu 1 \
        --clover-csw 1   --gtest_output=xml:gauge_path_test_${prec}.xml
```